notebooks/228-clip-zero-shot-image-classification/228-clip-zero-shot-image-classification.ipynb

```diff
@@ -0,0 +1,281 @@
+    "\n",
+    "![clip](https://raw.githubusercontent.com/openai/CLIP/main/CLIP.png)\n",
+    "\n",
+    "You can find more information about this model in the [research paper](https://arxiv.org/abs/2103.00020), [OpenAI blog](https://openai.com/blog/clip/), [model card](https://github.com/openai/CLIP/blob/main/model-card.md) and GitHub [repository](https://github.com/openai/CLIP).\n",
+    "\n",
+    "In this notebook, we will use [openai/clip-vit-base-patch16](https://huggingface.co/openai/clip-vit-base-patch16), which available via HuggingFace Transformers library, but the same steps aplicable for other models CLIP family.\n",
+    "\n",
+    "First, we need to create `CLIPModel` class object and initialize it with model configuration and weights using `from_pretrained` method. Model will be automatically downloaded from huggingface hub and cached for next usages.\n",
+    "`CLIProcessor` class is wrapper for input data preprocessing. It includes both encode the text using tokenizer and prepare the images."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from transformers import CLIPProcessor, CLIPModel\n",
+    "\n",
+    "# load pretrained model\n",
+    "model = CLIPModel.from_pretrained(\"openai/clip-vit-base-patch16\")\n",
+    "# load preprocessor for model input\n",
+    "processor = CLIPProcessor.from_pretrained(\"openai/clip-vit-base-patch16\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from typing import List\n",
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "from PIL import Image\n",
+    "\n",
+    "\n",
+    "def visualize_result(image:Image, labels:List[str], probs:np.ndarray, top:int = 5):\n",
+    "    \"\"\" \n",
+    "    Utility function for visualization classuification results\n",
+    "    params:\n",
+    "      image: input image\n",
+    "      labels: list of classification labels\n",
+    "      probs: model predicted softmaxed probabilities for each label\n",
+    "      top: number of the highest probability results for visualization\n",
+    "    returns:\n",
+    "      None\n",
+    "    \"\"\"\n",
+    "    plt.figure(figsize=(64, 64))\n",
+    "    top_labels = np.argsort(-probs)[:min(top, probs.shape[0])]\n",
+    "    top_probs = probs[top_labels]\n",
+    "    plt.subplot(8, 8, 1)\n",
+    "    plt.imshow(image)\n",
+    "    plt.axis(\"off\")\n",
+    "\n",
+    "    plt.subplot(8, 8, 2)\n",
+    "    y = np.arange(top_probs.shape[-1])\n",
+    "    plt.grid()\n",
+    "    plt.barh(y, top_probs)\n",
+    "    plt.gca().invert_yaxis()\n",
+    "    plt.gca().set_axisbelow(True)\n",
+    "    plt.yticks(y, [labels[index] for index in top_labels])\n",
+    "    plt.xlabel(\"probability\")   "
+   ]
+  },
```

Comment:
Without running notebook it is hard to predict what will visualization will do, I guess it is normal, just unfortunately hard to understand from the code.
Tutorials probably will have this executed already, not sure if image will be shown there. Later it is also not visible.

Reply:
it is requirement do not store cells output for making ci working. I have plan to update ci pipeline to remove them automatically as part of validation, then we can store image output results directly in notebooks